packages/fuels/src/cli/utils/getLatestFuelsVersion.ts

```diff
@@ -0,0 +1,25 @@
+import { checkAndLoadCache, saveToCache } from './fuelsVersionCache';
+
+export const getLatestFuelsVersion = async (): Promise<string | undefined> => {
+  const cachedVersion = checkAndLoadCache();
+  if (cachedVersion) {
+    return cachedVersion;
+  }
+
+  const data: { version: string } | null = await Promise.race([
+    new Promise((resolve) => {
+      setTimeout(() => resolve(null), 3000);
+    }),
+    fetch('https://registry.npmjs.org/fuels/latest').then((response) => response.json()),
+  ]);
+
+  if (!data) {
+    throw new Error('Failed to fetch latest fuels version.');
+  }
```

Comment:
Since an error is being thrown anyway, we can reject the promise instead of resolving it on line 11.